salt/minion.py

```diff
@@ -845,7 +863,7 @@ def _spawn_minions(self):
         Spawn all the coroutines which will sign in to masters
         '''
         masters = self.opts[u'master']
-        if self.opts[u'master_type'] == u'failover' or not isinstance(self.opts[u'master'], list):
+        if (self.opts[u'master_type'] == u'failover' or self.opts[u'master_type'] == u'distributed') or not isinstance(self.opts[u'master'], list):
```

Comment:
I like your approach a lot more, it is far more elegant, updated the commit. Thanks!